ivy/functional/ivy/general.py

```diff
@@ -556,31 +556,171 @@ def to_scalar(x: Union[ivy.Array, ivy.NativeArray]) -> Number:
+    >>> y = ivy.to_list(x)
+    >>> print(y)
+    [[[-1, 0, 1], [1, 0, -1]], [[1, -1, 0], [1, 0, -1]]]
+    >>> print(isinstance(y, list))
+    True
+    
+    With :code:`ivy.NativeArray` input:
+    
+    >>> x = ivy.native_array([-1, 0, 1])
+    >>> y = ivy.to_list(x)
+    >>> print(y)
+    [-1, 0, 1]
+    >>> print(isinstance(y, list))
+    True
+    
+    >>> x = ivy.native_array([[-1, 0, 1], \
+                              [-1, 0, 1], \
+                              [ 1, 0, -1]])
+    >>> y = ivy.to_list(x)
+    >>> print(y)
+    [[-1, 0, 1], [-1, 0, 1], [1, 0, -1]]
+    >>> print(isinstance(y, list))
+    True
+    
+    >>> x = ivy.native_array([[[-1, 0, 1], \
+                               [1, 0, -1]], \
+                              [[1, -1, 0], \
+                               [1, 0, -1]]])
+    >>> y = ivy.to_list(x)
+    >>> print(y)
+    [[[-1, 0, 1], [1, 0, -1]], [[1, -1, 0], [1, 0, -1]]]
+    >>> print(isinstance(y, list))
+    True
+    
+    With a mix of :code:`ivy.Container` and :code:`ivy.Array` input:
+    
+    >>> x = ivy.Container(a=ivy.array([-1, 0, 1]))
+    >>> y = ivy.to_list(x)
+    >>> print(y)
+    {
+        a: [-1, 0, 1]
+    }
+    
+    >>> x = ivy.Container(a=ivy.array([[-1, 0, 1], \
+                                       [-1, 0, 1], \
+                                       [1, 0, -1]]))
+    >>> y = ivy.to_list(x)
+    >>> print(y)
+    {
+        a: [[-1, 0, 1], [-1, 0, 1], [1,0,-1]]
+    }
+    
+    >>> x = ivy.Container(a=ivy.array([[[-1, 0, 1], \
+                                        [1, 0, -1]], \
+                                       [[1, -1, 0], \
+                                        [1, 0, -1]]])
+    >>> y = ivy.to_list(x)
+    >>> print(y)
+    {
+        a: [[[-1, 0, 1], [1, 0, -1]], [[1, -1, 0], [1, 0, -1]]]
+    }
+    
+    With a mix of :code:`ivy.Container` and :code:`ivy.NativeArray` input:
+    
+    >>> x = ivy.Container(a=ivy.native_array([-1, 0, 1]))
+    >>> y = ivy.to_list(x)
+    >>> print(y)
+    {
+        a: [-1, 0, 1]
+    }
+    
+    >>> x = ivy.Container(a=ivy.native_array([[-1 0 1], \
+                                              [-1 0 1], \
+                                              [1 0 -1]])
+    >>> y = ivy.to_list(x)
+    >>> print(y)
+    {
+        a: [[-1, 0, 1], [-1, 0, 1], [1, 0, -1]]
+    }
+    
+    >>> x = ivy.Container(a=ivy.native_array([[[-1 0 1], \
+                                               [1 0 -1]], \
+                                              [[1 -1 0], \
+                                               [1 0 -1]]]))
+    >>> y = ivy.to_list(x)
+    >>> print(y)
+    {
+        a: [[[-1, 0, 1], [1, 0, -1]], [[1, -1, 0], [1, 0, -1]]]
+    }
+
+    Instance Method Examples
+    ------------------------
+
+    With :code:`ivy.Array` instance method:
+
+    >>> x = ivy.array([0, 1, 2])
+    >>> y = x.to_list()
+    >>> print(y)
+    ivy.to_list([0, 1, 2])
+
+    With :code:`ivy.NativeArray` instance method:
```

Comment:
Hi @husseinalamutu ,
I just made some changes to the format of your output statements since they should be in the form `[-1, 0, 1]` and not `ivy.to_list([-1, 0, 1])` for example.

Last change to make: `ivy.NativeArray` doesn't work as an instance method here so you can remove the 3 examples at lines 694, 701 and 710, then replace these with just 1 example using only `ivy.Container` as an instance method.

Once you've made this change I can merge all your edits in.